Week2/TriplateSum.java

```diff
@@ -0,0 +1,81 @@
+package sum3;
+
+import java.util.*;
+
+public class TriplateSum 
+{
+	public static Set<Set<Integer>> Sum3(int ar[])
+	{
+		 Set<Set<Integer>> bags = new HashSet<>();
+		 int ans[]= {0},n=ar.length,left,right;
+		 if(ar == null || ar.length<3) //array is NULL or less than two elements;

```

Comment:
This check needs to happen before line 10, otherwise the call to ar.length will throw an exception is ar is null. You should always have tests for your error cases to make sure they work as well